jinahub/indexers/searcher/HnswlibSearcher/hnswlib_searcher.py

```diff
@@ -164,7 +167,19 @@ def index(
         if len(docs_to_update) == 0:
             return
 
-        embeddings = docs_to_update.embeddings
+        try:
+            embeddings = docs_to_update.embeddings
+        except (TypeError, ValueError) as ex:
+            if self.ignore_invalid_docs:
+                docs_to_update = docs_to_update.traverse_flat(
+                    ['r'], filter_fn=lambda d: d.embedding is not None
```

Comment:
and `d.embedding.shape == self.blablabla.shape`, no?

Reply:
I would say the mis-matched shape can directly raise an exception.

Reply:
Yesterday we agreed that Indexer should not implement any filtering. It is the job of the upstream to make sure documents are correct. So this PR should be closed, right?

Reply:
yes, we can close this PR.